src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -958,6 +968,8 @@ class MergeTreeData : public IStorage, public WithMutableContext
     /// Remove local files and remote files if needed
     virtual bool removeDetachedPart(DiskPtr disk, const String & path, const String & part_name, bool is_freezed);
 
+    virtual String getTableSharedID() const { return ""; }
```

Comment:
Maybe throw `NOT_IMPLEMETED` by default?

Reply:
No, MutateTask use it.

Reply:
But why MutateTask use it for non-replicated MergeTree?